File: src/main/scala/com/github/mauricio/async/db/postgresql/column/TimeWithTimezoneEncoderDecoder.scala
package com.github.mauricio.postgresql.column

import org.joda.time.format.DateTimeFormat

/**
 * User: Maurício Linhares
 * Date: 3/11/12
 * Time: 5:35 PM
 */

object TimeWithTimezoneEncoderDecoder extends TimeEncoderDecoder {

  private val format = DateTimeFormat.forPattern("HH:mm:ss.SSSSSSZ")

  override def formatter = format

}

File: db-async-common/src/main/java/com/github/jasync/sql/db/column/TimeWithTimezoneEncoderDecoder.kt
package com.github.jasync.sql.db.column

import com.github.jasync.sql.db.util.microsecondsFormatter
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeFormatterBuilder

object TimeWithTimezoneEncoderDecoder : TimeEncoderDecoder() {

    private val format = DateTimeFormatterBuilder().appendPattern("HH:mm:ss")
        .appendOptional(microsecondsFormatter)
        .appendPattern("[X][Z]")
        .toFormatter()

    override fun formatter(): DateTimeFormatter = format
}
